### DISCUSSION TITLE
What does let g:ale_completion_autoimport means?

### DISCUSSION THREAD

**Discussion by ubaldot** [w0rp/ale#4077] [Q&A - Ask for help with problems] [answered]
The documentation is fairly cryptic and it is difficult for me to understand what is it. 
Could someone please explain? 
Possible by referring to a C project as an example.

Thanks! :)

**Comment by hsanson** [comment, marked as answer]
If you use a LSP that supports completions (e.g. eclipselsp for Java) then whenever you get auto-completion results for a class you do not have imported, then ALE will automatically add the import to your file. See in the example below how the LocalDate class is autoimported after selecting an autocompletion item:

![autoimport](https://user-images.githubusercontent.com/900716/154688035-9eadb16c-0611-4497-8704-42eb010b327f.gif)


Note again that this works only if you have enabled a LSP linter for your language (C?) that support autocompletion with extra edit text attributes.  I know eclipselsp does but have no idea about C LSP linters.
